src/runtime/particle-execution-host.ts

```diff
@@ -283,18 +285,31 @@ class PECOuterPortImpl extends PECOuterPort {
   }
 
   async onGetBackingStore(callback: number, storageKey: string, type: Type) {
-    if (!storageKey) {
-      // XXX
-      storageKey = this.arc.storageProviderFactory.baseStorageKey(type, this.arc.storageKey as string || 'volatile');
-    }
-    const store = await this.arc.storageProviderFactory.baseStorageFor(type, storageKey);
+    let store;
+    if (Flags.useNewStorageStack) {
+      if (!storageKey) {
+        // TODO(shanestephens): What should we do here?!
+        throw new Error(`Don't know how to invent new storage keys for new storage stack when we only have type information`);
+      }
+      const key = StorageKeyParser.parse(storageKey);
+      // TODO(shanestephens): We could probably register the active store here, but at the moment onRegister and onProxyMessage both
+      // expect to be able to do activation
+      const storeBase = new Store({id: storageKey, exists: Exists.MayExist, storageKey: key, type});
+      this.GetBackingStoreCallback(storeBase, callback, type, type.toString(), storageKey, storageKey);
+    } else {
+      if (!storageKey) {
+        // XXX
```

Comment:
?

Reply:
This was in the old code too.. :( It's gonna get deleted once we switch across to the new stack.